src/tree/registries/auth/oAuthUtils.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { URL } from 'url';
+import { parseError } from 'vscode-azureextensionui';
+import { IOAuthContext } from './IAuthHelper';
+
+const realmRegExp = /realm=\"([^"]+)\"/i;
+const serviceRegExp = /service=\"([^"]+)\"/i;
+const scopeRegExp = /scope=\"([^"]+)\"/i;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/tslint/config
+export function getWwwAuthenticateHeader(error: any): IOAuthContext {

```

Comment:
Is this specific to the Docker V2 API, or a more general OAuth mechanism?

Reply:
The `Www-Authenticate` header is supposed to be a generic OAuth thing, so I put it over here.